subsys/bluetooth/audio/vocs_client.c

```diff
@@ -0,0 +1,688 @@
+		inst->cli.end_handle = chrc->value_handle;
+
+		if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_STATE)) {
+			BT_DBG("Volume offset state");
+			inst->cli.state_handle = chrc->value_handle;
+			sub_params = &inst->cli.state_sub_params;
+		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_LOCATION)) {
+			BT_DBG("Location");
+			inst->cli.location_handle = chrc->value_handle;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				sub_params = &inst->cli.location_sub_params;
+			}
+			if (chrc->properties & BT_GATT_CHRC_WRITE_WITHOUT_RESP) {
+				inst->cli.location_writable = true;
+			}
+		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_CONTROL)) {
+			BT_DBG("Control point");
+			inst->cli.control_handle = chrc->value_handle;
+		} else if (!bt_uuid_cmp(chrc->uuid, BT_UUID_VOCS_DESCRIPTION)) {
+			BT_DBG("Description");
+			inst->cli.desc_handle = chrc->value_handle;
+			if (chrc->properties & BT_GATT_CHRC_NOTIFY) {
+				sub_params = &inst->cli.desc_sub_params;
+			}
+			if (chrc->properties & BT_GATT_CHRC_WRITE_WITHOUT_RESP) {
+				inst->cli.desc_writable = true;
+			}
+		}
+
+		if (sub_params) {
+			int err;
+
+			sub_params->value = BT_GATT_CCC_NOTIFY;
+			sub_params->value_handle = chrc->value_handle;
+			/*
+			 * TODO: Don't assume that CCC is at handle + 2;
+			 * do proper discovery;
+			 */
+			sub_params->ccc_handle = attr->handle + 2;
+			sub_params->notify = vocs_client_notify_handler;
+			err = bt_gatt_subscribe(conn, sub_params);
+
+			if (err) {
+				BT_WARN("Could not subscribe to handle %u",
+					sub_params->ccc_handle);
+			}
+		}
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+int bt_vocs_client_state_get(struct bt_conn *conn, struct bt_vocs *inst)
+{
+	int err;
+
+	CHECKIF(!conn) {
+		BT_DBG("NULL conn");
+		return -ENOTCONN;
+	}
+
+	CHECKIF(!inst) {
+		BT_DBG("NULL instance");
+		return -EINVAL;
+	}
+
+	if (!inst->cli.state_handle) {
+		BT_DBG("Handle not set");
+		return -EINVAL;
+	}
+
+	if (inst->cli.busy) {
+		BT_DBG("Handle not set");
+		return -EBUSY;
+	}
+
+	inst->cli.read_params.func = vocs_client_read_offset_state_cb;
+	inst->cli.read_params.handle_count = 1;
+	inst->cli.read_params.single.handle = inst->cli.state_handle;
+	inst->cli.read_params.single.offset = 0U;
+
+	err = bt_gatt_read(conn, &inst->cli.read_params);
+	if (!err) {
+		inst->cli.busy = true;
+	}
+	return err;
+}
+
+int bt_vocs_client_location_set(struct bt_conn *conn, struct bt_vocs *inst, uint32_t location)
+{
+	int err;
+
+	CHECKIF(!conn) {
+		BT_DBG("NULL conn");
+		return -ENOTCONN;
+	}
+
+	CHECKIF(!inst) {
+		BT_DBG("NULL instance");
+		return -EINVAL;
+	}
+
+	if (!inst->cli.location_handle) {
+		BT_DBG("Handle not set");
+		return -EINVAL;
+	} else if (inst->cli.busy) {
+		return -EBUSY;
+	} else if (!inst->cli.location_writable) {
+		BT_DBG("Location is not writable on peer service instance");
+		return -EPERM;
+	}
+
+	err = bt_gatt_write_without_response(conn, inst->cli.location_handle,
+					     &location, sizeof(location), false);
+
+	return err;
```

Comment:
Seems like you can probably remove `err` completely and just do `return bt_gatt_write_without_response(...);`